spring-cloud-app-broker-acceptance-tests/src/main/java/org/springframework/cloud/appbroker/acceptance/AppBrokerApplication.java

```diff
@@ -74,13 +77,23 @@ public DeleteServiceInstanceWorkflow deleteServiceInstanceWorkflow() {
 	}
 
 	/**
-	 * A no-op ServiceInstanceBindingService bean
+	 * A parameters transformer bean that conditionally swap out backing service
 	 *
 	 * @return the bean
 	 */
 	@Bean
-	public ServiceInstanceBindingService serviceInstanceBindingService() {
-		return new NoOpServiceInstanceBindingService();
+	public ParametersTransformerFactory<BackingService, ?> conditionalNewBackingService() {
+		return new ConditionalNewBackingService();
+	}
+
+	/**
+	 * A parameters transformer bean that conditionally swap out backing service
+	 *
+	 * @return the bean
+	 */
+	@Bean
+	public ParametersTransformerFactory<BackingApplication, ?> conditionalBackingAppUpdate() {
+		return new ConditionalBackingAppUpdate();
```

Comment:
I am wondering a couple of things:
- Can this be part of a create or update workflow so that the example is more realistic?
- Can this be part of the test instead of available for all the tests so that it's scoped only for those tests?

Reply:
Iterating over my thoughts, we need a previous state and then apply this new state on an update. So even a workflow might be confusing, wondering if there is something else we can do.